kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -123,7 +123,7 @@ private[kyo] class IOPromise[+E, +A](init: State[E, A]) extends Safepoint.Interc
                 case l: Linked[E, A] @unchecked =>
                     onCompleteLoop(l.p)
                 case v =>
-                    IOPromise.eval(f, v.asInstanceOf[Result[E, A]])
```

Comment:
The `eval` method was also introduced recently. I'm not sure why since it's inlined but it was introducing a function allocation